awx/main/scheduler/task_manager.py

```diff
@@ -597,27 +571,23 @@ def reap_jobs_from_orphaned_instances(self):
                 logger.error(f'{j.execution_node} is not a registered instance; reaping {j.log_format}')
                 reap_job(j, 'failed')
 
-    def calculate_capacity_consumed(self, tasks):
-        self.graph = InstanceGroup.objects.capacity_values(tasks=tasks, graph=self.graph)
-
-    def consume_capacity(self, task, instance_group, instance=None):
-        logger.debug(
-            '{} consumed {} capacity units from {} with prior total of {}'.format(
-                task.log_format, task.task_impact, instance_group, self.graph[instance_group]['consumed_capacity']
-            )
-        )
-        self.graph[instance_group]['consumed_capacity'] += task.task_impact
-        for capacity_type in ('control', 'execution'):
-            if instance is None or instance.node_type in ('hybrid', capacity_type):
-                self.graph[instance_group][f'consumed_{capacity_type}_capacity'] += task.task_impact
-
-    def get_remaining_capacity(self, instance_group, capacity_type='execution'):
-        return self.graph[instance_group][f'{capacity_type}_capacity'] - self.graph[instance_group][f'consumed_{capacity_type}_capacity']
+            # TODO: Test with container group jobs where the control pod in k8s is deleted/goes away
+            # This may not actually catch if the control node is lost, because j.controller_node may still be set. We might need
+            # to compare against a list of current instance hostnames
+            if (
+                settings.IS_K8S
+                and j.is_container_group_task
+                and not isinstance(j, WorkflowJob)
+                and not j.controller_node
+                and j.status in ['waiting', 'running']
+            ):
+                logger.error(f'Job {j.id} has no control node, but is in status {j.status}. The control node pod must have been lost; reaping {j.log_format}')
+                reap_job(j, 'error')
```

Comment:
Reaping a job at this location seems very suspect to me. How could the control node have gone away? You just ran in a loop to select it. This doesn't add up to me at all.